yt_dlp/extractor/joqrag.py

```diff
@@ -0,0 +1,54 @@
+import urllib.parse
+
+from .common import InfoExtractor
+
+
+class JoqrAgIE(InfoExtractor):
+    IE_DESC = '超!A&G+ 文化放送 Nippon Cultural Broadcasting, Inc. (JOQR)'
+    _VALID_URL = r'''(?x)
+                    (https?://www\.uniqueradio\.jp/agplayer5/player\.php)|
+                    (https?://www\.uniqueradio\.jp/agplayer5/inc-player-hls\.php)|
+                    (https?://(?:www\.)?joqr\.co\.jp/ag/)|
+                    (https?://(?:www\.)?joqr\.co\.jp/qr/(?:agdailyprogram|agregularprogram)/)
+                    '''
+    _TESTS = [{
+        'url': 'https://www.uniqueradio.jp/agplayer5/player.php',
+        'info_dict': {
+            'id': 'live',
+            'title': str,
+            'description': str,
+            'live_status': 'is_live',
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = 'live'
+
+        metadata = self._download_webpage(
+            'https://www.uniqueradio.jp/aandg', video_id,
+            note='Downloading metadata', errnote='Failed to download metadata')
+        title = urllib.parse.unquote_plus(
+            self._search_regex(r'var\s+Program_name\s*=\s*["\']([^"\']+)["\']', metadata, 'program title'))
+        desc = urllib.parse.unquote_plus(
+            self._search_regex(r'var\s+Program_text\s*=\s*["\']([^"\']+)["\']', metadata, 'program description'))
+
+        m3u8_path = self._search_regex(
+            r'<source\s[^>]*\bsrc="([^"]+)"',
+            self._download_webpage(
+                'https://www.uniqueradio.jp/agplayer5/inc-player-hls.php', video_id,
+                note='Downloading player data', errnote='Failed to download player data'),
+            'm3u8 url')
+        formats = self._extract_m3u8_formats(
+            f'https://www.uniqueradio.jp/{m3u8_path}', video_id, fatal=False)
+
+        return {
+            'id': video_id,
+            'title': f'{title} - 超!A&G+',
```

Comment:
personally I'd have the title as just `title`, and put `超!A&G+` in `channel` (and/or `uploader`)
```suggestion
            'title': title,
            'channel': '超!A&G+',
```
and if i still want 超!A&G+ in the filename i can do `-o "%(title)s - %(channel)s [%(id)s].%(ext)s"`